core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java

```diff
@@ -81,85 +90,103 @@ final class JdbcUtil {
           + TABLE_NAME
           + ")"
           + ")";
-  static final String GET_TABLE_SQL =
+  static final String UPDATE_CATALOG_SQL =
+      "ALTER TABLE " + CATALOG_TABLE_VIEW_NAME + " ADD COLUMN " + TYPE + " VARCHAR(5)";
+
+  private static final String GET_SQL =
       "SELECT * FROM "
-          + CATALOG_TABLE_NAME
+          + CATALOG_TABLE_VIEW_NAME
           + " WHERE "
           + CATALOG_NAME
           + " = ? AND "
           + TABLE_NAMESPACE
           + " = ? AND "
           + TABLE_NAME
-          + " = ? ";
-  static final String LIST_TABLES_SQL =
+          + " = ? AND "
+          + TYPE
+          + " = ?";
+  private static final String GET_TABLE_SQL =
+      GET_SQL
+          + " OR "
+          + TYPE
+          + " IS NULL"; // type is null when the SQL database has been updated from previous version
+  static final String LIST_SQL =
       "SELECT * FROM "
-          + CATALOG_TABLE_NAME
+          + CATALOG_TABLE_VIEW_NAME
           + " WHERE "
           + CATALOG_NAME
           + " = ? AND "
           + TABLE_NAMESPACE
+          + " = ? AND "
+          + TYPE
           + " = ?";
-  static final String RENAME_TABLE_SQL =
+  static final String LIST_TABLE_SQL = LIST_SQL + " OR " + TYPE + " IS NULL";
+  static final String RENAME_SQL =
       "UPDATE "
-          + CATALOG_TABLE_NAME
+          + CATALOG_TABLE_VIEW_NAME
           + " SET "
           + TABLE_NAMESPACE
-          + " = ? , "
+          + " = ?, "
           + TABLE_NAME
-          + " = ? "
+          + " = ?"
           + " WHERE "
           + CATALOG_NAME
           + " = ? AND "
           + TABLE_NAMESPACE
           + " = ? AND "
           + TABLE_NAME
-          + " = ? ";
-  static final String DROP_TABLE_SQL =
+          + " = ? AND "
+          + TYPE
+          + " = ?";
+  static final String RENAME_TABLE_SQL = RENAME_SQL + " OR " + TYPE + " IS NULL";
```

Comment:
same as above, I wonder if it wouldn't be better to have the full query here with `AND (... TYPE = ? OR TYPE IS NULL)`

Reply:
Yeah, it would be probably cleaner. I will do that.